serviceaccount/reconciler.go

```diff
@@ -0,0 +1,143 @@
+package serviceaccount
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/to"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/giantswarm/workload-identity-operator-gcp/webhook"
+)
+
+const (
+	AnnotationSecretMetadata  = "kubernetes.io/service-account.name" //#nosec G101
+	AnnotationSecretManagedBy = "app.kubernetes.io/managed-by"       //#nosec  G101
+
+	SecretManagedBy = "workload-identity-operator-gcp" //#nosec G101
+
+	SecretNameSuffix = "google-application-credentials" //#nosec G101
+)
+
+type ServiceAccountReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Logger logr.Logger
+}
+
+//+kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=serviceaccounts/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core,resources=serviceaccounts/finalizers,verbs=update
+
+func (r *ServiceAccountReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	serviceAccount := &corev1.ServiceAccount{}
+
+	err := r.Client.Get(ctx, req.NamespacedName, serviceAccount)
```

Comment:
Nitpick: `Client` can be omitted here, since it's embedded.

Reply:
Sorry I didn't make it clear, but I meant all the places where `r.Client.X()` is used